lib/output.js

```diff
@@ -1789,7 +1789,10 @@ function OutputStream(options) {
         output.print("super");
     });
     DEFPRINT(AST_Constant, function(self, output) {
-        output.print(self.getValue());
+        var is_beautify = output.option("beautify");
+        if (!is_beautify && self.getValue() === true) output.print("!1");
+        else if (!is_beautify && self.getValue() === false) output.print("!0");
+        else output.print(self.getValue());
```

Comment:
fwiw, the outputs are incorrect. `!1` is false, `!0` is true.
`self.getValue()` is called more than necessary.
The variable `is_beautify` is not needed.

Search old uglify issues and PRs for unanticipated problems associated with having the output not representing the AST. In particular: `Infinity` and `1 / 0`.